planemo/commands/cmd_container_register.py

```diff
@@ -74,30 +74,40 @@ def cli(ctx, paths, **kwds):
         ctx.vlog("Handling conda_targets [%s]" % conda_targets)
         mulled_targets = conda_to_mulled_targets(conda_targets)
         mulled_targets_str = "- " + "\n- ".join(map(conda_build_target_str, mulled_targets))
-        if len(mulled_targets) < 2:
-            ctx.vlog("Skipping registeration, fewer than 2 targets discovered.")
-            # Skip these for now, we will want to revisit this for conda-forge dependencies and such.
-            continue
-
         best_practice_requirements = True
         for conda_target in conda_targets:
             best_hit, exact = best_practice_search(conda_target, conda_context=conda_context)
-            if not best_hit or not exact:
-                ctx.vlog("Target [%s] is not available in best practice channels - skipping" % conda_target)
+            if not best_hit:
+                ctx.log("Target [%s] is not available in best practice channels - skipping" % conda_target)
                 best_practice_requirements = False
+            if not exact:
+                ctx.log("Target version [%s] for package [%s] is not available in best practice channels - please specify the full version",
+                        conda_target.version,
+                        conda_target.package)
 
         if not best_practice_requirements:
             continue
 
-        name = v2_image_name(mulled_targets)
-        tag = "0"
-        name_and_tag = "%s-%s" % (name, tag)
-        target_filename = os.path.join(registry_target.output_directory, "%s.tsv" % name_and_tag)
-        ctx.vlog("Target filename for registeration is [%s]" % target_filename)
-        if os.path.exists(target_filename):
-            ctx.vlog("Target file already exists, skipping")
+        if len(mulled_targets) < 1:
+            ctx.log("Skipping registration, no targets discovered.")
             continue
 
+        name = v2_image_name(mulled_targets)
+
+        if len(mulled_targets) == 1:
+            target_filename = os.path.join(registry_target.output_directory, "hash.tsv")
```

Comment:
@mvdbeek what about putting those also into separate files. Like here: https://github.com/BioContainers/multi-package-containers/pull/795/files

In this way we really can distinguish bettwen manually added containers and automatic ones.

Reply:
I'm not in favor of this. Much easier to just parse one file, no chance of duplication. If there is a reason to discriminate automatically built from not automatically built I don't think the difference should be a single file vs in one big file.

Reply:
You don't need to parse a file, you look up the path as it is done already with the multi-containers and like it is working for years. If the file is there, skip it. No parsing needed imho.

Reply:
We should be parsing the file anyway to check if the build already exists. This also avoids any potential character issue (utf-8 on non-utf8 systems, or other fairly arbitrary character restrictions on samba/cifs. That wasn't an issue with multi-containers because of the hash.

Reply:
I mean it doesn't matter too much. j think a single file is more user-friendly, we can always revisit this later if it is really an issue. I'll update the code.

Reply:
Just updated it, thanks @bgruening!